examples/ex30_proposed.cpp

```diff
@@ -0,0 +1,466 @@
+//                                MFEM Example 30
+//
+// Compile with: make ex30
+//
+// Sample runs:  ex30 -m ../data/inline-segment.mesh -o 2
+//               ex30 -m ../data/hexagon.mesh -o 2
+//               ex30 -m ../data/star.mesh -o 2
+//               ex30 -m ../data/fichera.mesh -o 3 -r 1
```

Comment:
Although the `../data/amr-hex.mesh -r 1` and `ex30 -m ../data/amr-hex.mesh -r 1` execute without errors, I am not sure whether they should be here. I thought that this basis was meant for 1D and 2D domains. On that note, a sample run where the domain is an embedded 2D surface may be interesting. Thoughts?

Reply:
@mlstowell Any thoughts on this?